### ISSUE TITLE
Disable executable file icon/color

### ISSUE THREAD

**Issue by drzhivago420** [gokcehan/lf#992]
I've tried moving around the priority and outright commenting the ex line in both the `icons` and `colors` files, with no luck.
Is there any way for me to actually see any icon or color aside from green and rockets besides "fixing my permissions"?

**Comment by ilyagr** [comment]
If I understand the question correctly, instead of commenting out the line, try changing it so that it has the same color/icon as the style for "normal" files.

**Comment by drzhivago420** [comment]
I want stuff to be like `b`
Currently it is like `a`
Without changing permissions.
![executable](https://user-images.githubusercontent.com/110633694/200330598-aba72624-f193-45d2-bed8-f0671592ca8f.png)

**Comment by drzhivago420** [comment]
I understand why it works like that for colors, but at least for file icons, I wish it ignored if it's executable or not. I'm tired of seeing that dumb rocket.

**Comment by nhat-vo** [comment]
Yes indeed I have the same problem. It seems that #506 was supposed to fix this issue by prioritizing pattern matching over file type. But in my case I still find that `ex` is preferred over `*.py` like below
![image](https://user-images.githubusercontent.com/30724347/205436209-cab0fd07-77e9-4d3f-8265-4dcec83a7e5e.png)

**Comment by jpmvferreira** [comment]
I have precisely the same issue.
Why not take the simpler approach and have the order of the configuration file matter when it comes to applying a specific icon/color? This would allow different users to sort differently according to their taste, perhaps some actually like having their icon overwritten when the file is an executable.
This way, this issue could be easily solved by placing the executable icon in the bottom of the file, acting as a fallback if no other match was triggered.

**Comment by Tanish2002** [comment]
+1 for this feature.

**Comment by ted1277** [comment]
Any solution for this problem? I'm using lf in WSL, so all Windows files are "executable files". I've set `EXEC 0` in `.dircolors`, so `ls` prints colored files according to the extensions of files. However, lf behaves differently, files have no colors in lf.

**Comment by hakan-demirli** [comment]
I delete this part and get rid of all executable icons:
https://github.com/gokcehan/lf/blob/beedda2d6708259b03cc6921e0c1cdde231a2031/icons.go#L139-L141
A terrible workaround but considering I rarely interact with executable files over `lf` I haven't felt their absence.

**Comment by CatsDeservePets** [comment]
Related:
- https://github.com/gokcehan/lf/discussions/1875
- https://github.com/gokcehan/lf/issues/671

**Comment by joelim-work** [comment]
Judging from the history of the code, it looks the behavior was just copied from `ls`/`LS_COLORS`, which also prioritizes filetype over extension and does not allow the user to specify otherwise. It is an understandable decision, but it also does not work well on Windows when every file is marked as executable.

There are some possible ideas on what configuration can look like:

- A bool option like `iconspreferext` - simple but not very flexible
- A string option like `iconspriority` which indicates a priority list (e.g. `fullpath:extension:filetype`) - moderately flexible
- Have the `icons` file consist of a priority list of rules, as mentioned in https://github.com/gokcehan/lf/issues/992#issuecomment-1371364445 - very flexible, but might be less efficient (now have to compare against hundreds of rules for each file)

There were a couple of attempts at making the configuration more flexible (see #505 and #506), but they were turned down due to complexity and performance issues.

**Comment by CatsDeservePets** [comment]
> Judging from the history of the code, it looks the behavior was just copied from `ls`/`LS_COLORS`, which also prioritizes filetype over extension and does not allow the user to specify otherwise. It is an understandable decision, but it also does not work well on Windows when every file is marked as executable.

Some days ago, I merged to following:
- https://github.com/gokcehan/lf/pull/2448

Under `Windows`, `lf` now uses the `PATHEXT` env var to determine whether a file is executable or not (which was already used to suggest program names inside command line completions). I think this makes more sense than checking the execute bit. It is also what the `dir` command in `PowerShell` uses for its coloured output.
However, while now getting less false positives (like when using `lf.exe` inside `WSL`, which you shouldn't lol), `Windows` users now will also see more `rockets` even though their custom icon for `.exe` files worked in the past.
Thinking about it, perhaps I should move it from `Fixed` to `Changed` in our changelog.

The rocket icon is something that annoys me as well. I compared some other tools that support `LS_COLORS` as well as custom icon and colour configuration: `joshuto`, `lsd`, `eza` (`ranger` has no official icon support and `yazi` only seems to support its own config system). While I don't know their exact matching order, they all had the following in common: using the extension based icon over the executable one while still using the executable colour. All except `lf`.

| lsd | eza | joshuto | lf |
|---|---|---|---|
| ![lsd](https://github.com/user-attachments/assets/fcd7e0ac-b525-46e2-b687-c19f9525a505) | ![eza](https://github.com/user-attachments/assets/2edf4598-5c71-4b5c-b018-605557689db8) | ![joshuto](https://github.com/user-attachments/assets/80410b89-43ae-4e21-aa36-cfb99eb0b405) | ![lf](https://github.com/user-attachments/assets/6df0fba2-bf86-4a01-a8ad-665692a1f94a) |

> There are some possible ideas on what configuration can look like:
> 
> - A bool option like `iconspreferext` - simple but not very flexible
> - A string option like `iconspriority` which indicates a priority list (e.g. `fullpath:extension:filetype`) - moderately flexible
> - Have the `icons` file consist of a priority list of rules, as mentioned in https://github.com/gokcehan/lf/issues/992#issuecomment-1371364445 - very flexible, but might be less efficient (now have to compare against hundreds of rules for each file)

I am fine with either option one or two. I doubt the third one would be worth the complexity and performance implications, specially considering that most issues regarding `lf`'s icon priority are specifically talking about the executable icon/rocket. It seems to be the common complaint across all of them.
